cmd/pipeline-controller/config_data_provider.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+type presubmitTests struct {
+	protected             []string
+	alwaysRequired        []string
+	conditionallyRequired []string
+}
+
+type ConfigDataProvider struct {
+	configGetter      config.Getter
+	updatedPresubmits map[string]presubmitTests
+	m                 sync.Mutex
+}
+
+func NewConfigDataProvider(configGetter config.Getter) *ConfigDataProvider {
+	provider := &ConfigDataProvider{
+		configGetter:      configGetter,
+		updatedPresubmits: make(map[string]presubmitTests),
+		m:                 sync.Mutex{},
+	}
+	provider.gatherData()
+	return provider
+}
+
+func (c *ConfigDataProvider) GetPresubmits(orgRepo string) presubmitTests {
+	c.m.Lock()
+	defer c.m.Unlock()
+	if presubmits, ok := c.updatedPresubmits[orgRepo]; ok {
+		return presubmits
+	}
+	return presubmitTests{}
+}
+
+func (c *ConfigDataProvider) Run() {
+	for {
+		time.Sleep(10 * time.Minute)
```

Comment:
What about using [Agent.Subscribe](https://github.com/kubernetes/test-infra/blob/1361dca58e78f2ac6ec6eda929d2c6f676e91b95/prow/config/agent.go#L385-L389) instead of polling each 10 mins?

Reply:
I understand the intention, but totally not needed here as we pool what is mounted to our local disk so no tokens of any sort are burned. In the initial version it was even 30 mins, but I guess anything above 5 mins is minimal load for the system.

Reply:
> no tokens of any sort are burned

My concern is about being more responsive to the configuration change events rather than consuming less tokens. It's not something I strictly require for this PR to get merged by the way. We can come back to it as soon as we receive complaints by the users.